Generalize TCP retxn docs to cover remote clusters (#74732)
elastic/elasticsearch: docs/reference/setup/sysconfig/tcpretries.asciidoc

[[system-config-tcpretries]]
=== TCP retransmission timeout

Each pair of {es} nodes communicates via a number of TCP connections which
<<long-lived-connections,remain open>> until one of the nodes shuts down or
communication between the nodes is disrupted by a failure in the underlying
infrastructure.

> TCP provides reliable communication over occasionally-unreliable networks by

Reviewer: Nit: No hyphen is needed between an adverb and adjective.
```suggestion
TCP provides reliable communication over occasionally unreliable networks by
```

hiding temporary network disruptions from the communicating applications. Your
operating system will retransmit any lost messages a number of times before
informing the sender of any problem. {es} must wait while the retransmissions
are happening and can only react once the operating system decides to give up.
Users must therefore also wait for a sequence of retransmissions to complete.

Most Linux distributions default to retransmitting any lost packets 15 times.
Retransmissions back off exponentially, so these 15 retransmissions take over
900 seconds to complete. This means it takes Linux many minutes to detect a
network partition or a failed node with this method. Windows defaults to just 5
retransmissions which corresponds with a timeout of around 6 seconds.

The Linux default allows for communication over networks that may experience
very long periods of packet loss but this default is excessive and even harmful
on the high quality networks used by most {es} installations. Highly-available

> clusters must be able to detect node failures promptly in order to react by

Reviewer: Nit: This sentence gets a bit long. I'd break it into two. Regardless, `in order to` can be shortened to `to`.
```suggestion
clusters must be able to detect node failures promptly. This lets the cluster react by
```
